dist/rules/no-internal-barrel-imports.js

```diff
@@ -154,10 +154,10 @@ const rule = {
 
         const thisModule = importNodeTs.getSourceFile();
 
-        const importInfo = ts.getResolvedModule(
+        const importInfo = program.getResolvedModule(
```

Comment:
is this backwards compatible for users who are using ts 5.2 and below?

Reply:
did not think to test that my bad. Will test on main with src code change and will report back

Reply:
On `main` branch of a fresh clone of repo, i change ts dep to 5.2.2 and copy/pasted the contents of this file to same file in main, then ran the test and it failed:
![image](https://github.com/iTwin/eslint-plugin/assets/98850418/dc1a0914-418c-46f3-81cf-eba03611db04)

It appears this change would not be compatible with 4.2.2 and lower ts versions. lmk what my next steps should be in this case. Thanks

Reply:
Perhaps check to see if `program.getResolvedModule` is indeed a function/defined and use it if so, otherwise default to `ts.getResolvedModule`

Reply:
That's a good idea, I have added that check as per your suggestion @ben-polinsky and a short comment explanation. I tested the src code on both 5.3.3 and 5.0.4 ts versions on this repo and both scenarios, tests are passing